bluez-async/src/lib.rs

```diff
@@ -458,33 +586,30 @@ impl BluetoothSession {
         ))
     }
 
-    /// Power on all Bluetooth adapters and start scanning for devices.
-    pub async fn start_discovery(&self) -> Result<(), BluetoothError> {
-        let bluez_root = Proxy::new(
-            "org.bluez",
-            "/",
-            DBUS_METHOD_CALL_TIMEOUT,
-            self.connection.clone(),
-        );
-        let tree = bluez_root.get_managed_objects().await?;
-        let adapters: Vec<_> = tree
-            .into_iter()
-            .filter_map(|(path, interfaces)| interfaces.get(ORG_BLUEZ_ADAPTER1_NAME).map(|_| path))
-            .collect();
-
+    /// Power on all Bluetooth adapters, set the given discovery filter, and then start scanning for
+    /// devices.
+    ///
+    /// Note that BlueZ combines discovery filters from all clients and sends events matching any
+    /// filter to all clients, so you may receive unexpected discovery events if there are other
+    /// clients on the system using Bluetooth as well.
+    ///
+    /// In most common cases, `DiscoveryFilter::default()` is fine.
+    pub async fn start_discovery(
+        &self,
+        discovery_filter: &DiscoveryFilter,
```

Comment:
As you already stated in the comment, in most cases the default filter is fine. Therefore, I suggest to keep the old `start_discovery()` but add a new method `start_discovery_with_filter()`. Then `start_discovery()` can look like this:

```rust
pub async fn start_discovery(&self) -> Result<(), BluetoothError> {
    self.start_discovery_with_filter(&Default::default)
}
```

This way you don't have to break the API or change the README.md .

Reply:
There's already a breaking change because of adding enum variants, but done.